modules/report/includes/commerce_pos_report.pages.inc

```diff
@@ -1005,3 +1004,300 @@ function commerce_pos_report_build_journal_role_table(array $filters = array())
+      $totals['item_sold'] += $net_sold_items;
+      $totals['unit_per_transaction'] += $unit_per_transaction;
+      $totals['dollar_per_transaction'] += $dollar_per_transaction;
+      $rows[] = array(
+        'data' => $table_row,
+        'class' => '',
+      );
+    }
+
+    $rows[] = array(
+      '',
+      '',
+      commerce_currency_format($totals['net_sales'], $currency_code),
+      $totals['transactions'],
+      $totals['item_sold'],
+      $totals['unit_per_transaction'],
+      commerce_currency_format($totals['dollar_per_transaction'], $currency_code),
+    );
+
+    $table = array(
+      '#theme' => 'table',
+      '#rows' => $rows,
+      '#header' => $header,
+      '#attached' => array(
+        'js' => array(
+          drupal_get_path('module', 'commerce_pos_report') . '/js/commerce_pos_report.journal_role.js',
+        ),
+        'library' => array(
+                array('system', 'drupal.ajax'),
+        ),
+      ),
+    );
+    return $table;
+  }
+  else {
+    return array(
+      '#markup' => t('No transactions found'),
+    );
+  }
+}
+
+/**
+ * Callback for the KPI form/report.
+ */
+function commerce_pos_report_kpi_report($form, &$form_state) {
+  global $user;
+  $form['#tree'] = TRUE;
+  $form['header'] = array(
+    '#markup' => theme('commerce_pos_header', array('account' => $user)),
+  );
+
+  $query_filter_defaults = array(
+    'results_per_page' => COMMERCE_POS_REPORT_KPI_REPORT_DEFAULT_ITEMS_PER_PAGE,
+  );
+
+  $query_params = drupal_get_query_parameters() + $query_filter_defaults;
+
+  $form['filters'] = array(
+    '#type' => 'container',
+  );
+
+  if (!isset($form_state['results_container_id'])) {
+    $form_state['results_container_id'] = 'commerce-pos-report-journal-role-container';
+  }
+
+  $js_settings = array(
+    'commercePosReport' => array(
+      'cssUrl' => url(drupal_get_path('module', 'commerce_pos_report') . '/css/commerce_pos_report_receipt.css', array(
+        'absolute' => TRUE,
+      )),
+    ),
+  );
+
+  $form['#attached']['libraries_load'][] = array('jquery-print');
+  $form['#attached']['css'][] = drupal_get_path('module', 'commerce_pos') . '/css/commerce_pos_style.css';
+  $form['#attached']['css'][] = drupal_get_path('module', 'commerce_pos_report') . '/css/commerce_pos_report.css';
+  $form['#attached']['js'][] = drupal_get_path('module', 'commerce_pos_report') . '/js/commerce_pos_report.journal_role.js';
+  $form['#attached']['js'][] = array(
+    'type' => 'setting',
+    'data' => $js_settings,
+  );
+  $form['report-options'] = array(
+    '#theme' => 'commerce_pos_report_options',
+  );
+  $form['options'] = array(
+    '#type' => 'container',
+    '#id' => 'commerce-pos-report-journal-role-options-container',
+  );
+  $form['options']['results_per_page'] = array(
+    '#type' => 'select',
+    '#title' => t('Items per page'),
+    '#options' => array(
+      25 => '25',
+      50 => '50',
+      100 => '100',
+      500 => '500',
+      -1 => t('all'),
```

Comment:
This is actually a bug that I fixed here: https://github.com/AcroMedia/commerce_pos/pull/46. Should be refactored here as well.